docs/manual/src/internals/foreign_calls.md

```diff
@@ -0,0 +1,88 @@
+# Rust -> Foreign calls
+
+How can Rust code make a call into the foreign language?
+The [lifting and lowering](./lifting_and_lowering.md) docs describe this at a high-level, this document will explore the details.
+
+This document only describes non-async calls.
+See the [Async FFI](./async-ffi.md) docs for a description of async calls.
+
+## VTables
+
+Calling into foreign code has 2 major differences from [calling Rust code](./rust_calls.md):
+
+* Foreign code is only called via an interface.
+* The Rust code does not have access to a library with functions it can call.
+
+Because of this, the foreign calls all go through a vtable that the foreign code registers with the Rust code.
```

Comment:
What does "registers" mean? Who does the registering? 

Is it: When UniFFI generates the foreign language template, it creates a VTable automatically with the functions Rust can call which are implemented on the foreign language through the build process in UniFFI?

Reply:
Yes that wasn't very clear.  "Register" here means the foreign code passes the callback to a generate Rust function so that the Rust code knows about it.  I just removed all of the wording, because the concept is shown in the code below and I think it's clearer there.